src/main/java/com/clevertap/maven/plugins/supertest/SuperTestMavenPlugin.java

```diff
@@ -125,6 +126,11 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                 }
                 classnameToTestcaseList.put(
                         runResult.getClassName(), runResult.getFailedTestCases());
+                testsToBeRetried += runResult.getFailedTestCases().size();
+            }
+
+            if (testsToBeRetried == 0) {
```

Comment:
This does not seem correct to me.

If there are no unit tests found, then `allTestClasses` will be empty. If it is not, then we have at least one unit test. And if we come to this point here, it says that there is no failing test, but in some cases, we don't get a xml file at all. E.g. if the test was running for too long and we had to interrupt the execution. In this case `testsToBeRetried` will be `0`, but we should actually retry.

The problem may be 

```
new TestListResolver(
                        includes, excludes, getTest(), testClassesDir).scanDirectories()
```

Check if `TestListResolver` works fine with non-trivial  `-Dtest=...`. If it works, just check `allTestClasses` in case mvn exited with non-zero code (i.e. potentially it had zero tests). Though to be even more precise, you may want to check `readProcessStdOut` for the right error message that there are no tests.